src/main/kotlin/io/zerobase/smarttracing/GraphDao.kt

```diff
@@ -216,4 +223,87 @@ class GraphDao(private val driver: Driver, val phoneUtil: PhoneNumberUtil) {
             }
         }
     }
+
+    /**
+     * Creates a user node and links to device id.
+     *
+     * @param name name of the user.
+     * @param phone phone of the user.
+     * @param email email of the user.
+     * @param id id of the device used to create it.
+     *
+     * @returns id of the user.
+     */
+    fun createUser(name: String?, phone: String?, email: String?, id: String): UserId? {
+        val validNum = if (phone != null) {
+            phoneUtil.isValidNumber(phoneUtil.parseAndKeepRawInput(phone, "US"))
+        } else {
+            true
+        }
+
+        if (!validNum) {
+            throw InvalidPhoneNumberException("Invalid phone number")
+        }
+
+        // Can cause issues as it throws an exception when device id was used
+        // to create another device
+        return driver.session().use {
+            it.writeTransaction { txn ->
+                val result = txn.run(
+                        """
+                            MATCH (d:Device { id: '${id}' })
+                            WHERE NOT () - [:OWNS] -> (d)
+                            CREATE (u: User {
+                                        id: '${UUID.randomUUID()}',
+                                        name: '${name}',
+                                        phone: '${phone}',
+                                        email: '${email}',
+                                        deleted: 'false'
+                            }) - [r:OWNS] -> (d)
+                            RETURN u.id as id
+                        """.trimIndent()
+                ).single()["id"].asString()
+                return@writeTransaction result?.let { UserId(it) }
+            }
+        }
+    }
+
+    /**
+     * "Deletes" the user
+     *
+     * @param id id of the user to delete
+     */
+    fun deleteUser(id: String) {
+         driver.session().use {
+            it.writeTransaction { txn ->
+                txn.run(
+                        """
+                        MATCH (u:User { id: '${id}' })
+                        SET u.deleted = 'true'
+                        """
+                )
+            }
+        }
+    }
+
+    /**
+     * Gets the user
+     *
+     * @param id the id of the user
+     *
+     * @return User struct
+     */
+    fun getUser(id: String): UserDaoStruct {
```

Comment:
this should use the `UserId` inline class as the parameter type. same with the other methods that take an id